### PR TITLE
Revert selecting virtual services

### PR DESCRIPTION
**Pull request by howardjohn**
Revert https://github.com/istio/istio/pull/33764 to fix https://github.com/istio/istio/issues/37691. Alternative is https://github.com/istio/istio/pull/37721.

This was primarily to get the new tests I wrote passing. These same tests pass on the release-1.11 branch, ensuring we have no behavioral regression. I am ok moving forward with https://github.com/istio/istio/pull/37721 if we pull in these tests there and they all pass (currently they do not).

### CODE REVIEW

**Comment by hzxuzhonghu** [comment]
The root cause is `SelectVirtualServices` is not right, if you define a sidecar with egress hosts set the bug is still same.

**Comment by howardjohn** [comment]
My understanding is this is reverting the SelectVS which is,as you
mentioned,not right. We can forward-fix it like your PR but it doesn't
handle all cases (see comment on your PR)

**Comment by hzxuzhonghu** [comment]
WIll import the tests

**Comment by howardjohn** [comment]
SGTM. I would obviously prefer your PR btw and don't want to merge this.
Was just to develop the tests

File: pilot/pkg/networking/core/v1alpha3/httproute_test.go [hunk 1]
```diff
@@ -1239,6 +1240,9 @@ func TestSidecarOutboundHTTPRouteConfig(t *testing.T) {
 				"test.com:8080": {
 					"test.com:8080": true, "8.8.8.8:8080": true,
 				},
+				"test-svc.testns.svc.cluster.local:80": {
+					"test-svc.testns.svc.cluster.local": true, "test-svc.testns.svc.cluster.local:80": true,
```
Comment (hzxuzhonghu) [inline pilot/pkg/networking/core/v1alpha3/httproute_test.go:1244]: which service is test-svc.testns.svc.cluster.local:80

Comment (hzxuzhonghu) [reply pilot/pkg/networking/core/v1alpha3/httproute_test.go:1244]: I wonder this is not right

**Comment by hzxuzhonghu** [comment]
https://github.com/istio/istio/pull/33764 is not the cause,  after https://github.com/istio/istio/pull/34247 it is broken.

**Review by GregHanson — APPROVED** [review]
waiting for final comments from @hzxuzhonghu

File: pilot/pkg/networking/core/v1alpha3/httproute.go [hunk 2]
```diff
@@ -215,11 +218,55 @@ func (configgen *ConfigGeneratorImpl) buildSidecarOutboundHTTPRouteConfig(
...
+
+			if match {
+				break
+			}
+		}
+
+		if match {
+			out = append(out, c)
+		} else {
+			log.Errorf("howardjohn: drop %v", c.Name)
```
Comment (hzxuzhonghu) [inline pilot/pkg/networking/core/v1alpha3/httproute.go:257]: drop

**Review by hzxuzhonghu — COMMENTED** [review]
LGTM

**Review by hzxuzhonghu — APPROVED** [review]

**Pull request merged**